### PR TITLE
chore(limit-count): don't return plugin inner error info to caller

### PR DESCRIPTION
**Pull request by okaybase**
### What this PR does / why we need it:

don't return limit-count inner error info(e.g. invalid redis password or invalid redis DB index) to caller

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/plugin/limit-count-redis2.t [hunk 1]
```diff
@@ -130,8 +130,6 @@ passed
 GET /hello
 --- error_code eval
 500
---- response_body
```
Comment (spacewander) [inline t/plugin/limit-count-redis2.t:133]: Need to check error_log

File: apisix/plugins/limit-count.lua [hunk 2]
```diff
@@ -188,7 +188,7 @@ function _M.access(conf, ctx)
         if conf.allow_degradation then
             return
         end
-        return 500, {error_msg = "failed to limit count: " .. err}
```
Comment (spacewander) [inline apisix/plugins/limit-count.lua:191]: Let's fix the message above: `core.log.error("failed to limit req: ", err)`

File: t/plugin/limit-count-redis2.t (hunk 1 above)
Comment (okaybase) [reply t/plugin/limit-count-redis2.t:133]: okay, thanks~ done~

File: apisix/plugins/limit-count.lua (hunk 2 above)
Comment (okaybase) [reply apisix/plugins/limit-count.lua:191]: okay, thanks~ done~

File: apisix/plugins/limit-count.lua [hunk 3]
```diff
@@ -184,11 +184,11 @@ function _M.access(conf, ctx)
             return conf.rejected_code
         end
 
-        core.log.error("failed to limit req: ", err)
+        core.log.error("failed to limit count: ", err)
         if conf.allow_degradation then
             return
         end
-        return 500, {error_msg = "failed to limit count: " .. err}
```
Comment (spacewander) [inline apisix/plugins/limit-count.lua:191]: For compatibility, we still need to return a `{error_msg = "..."}`

File: apisix/plugins/limit-count.lua [hunk 4]
```diff
@@ -184,11 +184,11 @@ function _M.access(conf, ctx)
...
+        return 500, {error_msg = "..."}
```
Comment (spacewander) [inline apisix/plugins/limit-count.lua:191]: Err, I should make it clear. What about `{error_msg = "failed to limit count"}`?

Comment (okaybase) [reply apisix/plugins/limit-count.lua:191]: okay, thanks~ done~

Comment (okaybase) [reply apisix/plugins/limit-count.lua]: okay, thanks~ done~

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**